packages/nitro-protocol/contracts/AssetHolder.sol

```diff
@@ -113,14 +115,17 @@ contract AssetHolder is IAssetHolder {
             );
         } 
 
-        // storage updated BEFORE asset transferred (prevent reentrancy)
 
+        // Event emitted regardless of success of external calls
+        emit AssetTransferred(fromChannelId, destination, affordsForDestination);
+
+        // storage updated BEFORE asset transferred (prevent reentrancy)
```

Comment:
It seems like this comment doesn't apply any longer? Do we understand why it's important to emit `AssetTransferred` after updating storage?

Reply:
I think the comment does apply, as it doesn't mention events. Regardless I do think it is slightly strange to have the event emitted before the assets are actually transferred, so I will move it to afterwards. Also, in the case of reentrancy, we could potentially get more events emitted than we should.

The more important thing is to do any external contract calls only after updating storage.